state/postgresql/v1/migrations.go

```diff
@@ -51,8 +56,20 @@ func performMigrations(ctx context.Context, db pginterfaces.PGXPoolConn, opts po
 				),
 			)
 			if err != nil {
-				return fmt.Errorf("failed to create state table: %w", err)
+				if pgErr, ok := err.(*pgconn.PgError); ok {
```

Comment:
Need to use `errors.As` because the error of type `pgconn.PgError` may be wrapped, so a simple type casting isn't enough. See example I linked above.